source/extensions/filters/network/zookeeper_proxy/filter.cc

```diff
@@ -216,13 +244,33 @@ void ZooKeeperFilter::onDecodeError() {
   setDynamicMetadata("opname", "error");
 }
 
-void ZooKeeperFilter::onRequestBytes(const uint64_t bytes) {
+void ZooKeeperFilter::onRequestBytes(const absl::optional<OpCodes> opcode, const uint64_t bytes) {
   config_->stats_.request_bytes_.add(bytes);
+
+  if (config_->enable_per_opcode_request_bytes_metrics_ && opcode.has_value()) {
+    if (*opcode == OpCodes::Connect) {
+      config_->stats_.connect_rq_bytes_.add(bytes);
+    } else {
+      ASSERT(config_->op_code_map_.contains(*opcode));
+      config_->op_code_map_[*opcode].rq_bytes_counter_->add(bytes);
+    }
+  }
+
   setDynamicMetadata("bytes", std::to_string(bytes));
 }
 
-void ZooKeeperFilter::onResponseBytes(const uint64_t bytes) {
+void ZooKeeperFilter::onResponseBytes(const absl::optional<OpCodes> opcode, const uint64_t bytes) {
   config_->stats_.response_bytes_.add(bytes);
+
+  if (config_->enable_per_opcode_response_bytes_metrics_ && opcode.has_value()) {
+    if (*opcode == OpCodes::Connect) {
+      config_->stats_.connect_resp_bytes_.add(bytes);
+    } else {
+      ASSERT(config_->op_code_map_.contains(*opcode));
```

Comment:
Same as above.

Reply:
This opcode comes from decoder.cc "decodeOnWrite" function. It should match the opcode decoded by the "decodeOnData" function.